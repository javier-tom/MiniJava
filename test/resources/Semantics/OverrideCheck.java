class Main {
    public static void main(String[] args) {
        System.out.println((new Override().run(10, 15)));
    }
}
class Override {
    int x;
    int y;
    boolean f1;
    boolean f2;

    public int run(int x, int y) {
        A a;
        B b;
        a = new A();
        b = new B();
        x = a.run(1,true);
        y = b.run(1,1);
        return 0;
    }
}

class A extends B {
    public int run(int x, boolean b) {
        return 2;
    }

    public C good() {
        return new C();
    }

    public F bad() {
        return new F();
    }
}

class B {
    public int run(int x, int b) {
        return 1;
    }

    public D good() {
        return new D();
    }

    public D bad() {
        return new D();
    }
}

class C extends D {}
class D extends F {}
class F {}