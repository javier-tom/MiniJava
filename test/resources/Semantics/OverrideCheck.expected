class A:
  super: B
  fields:
  method run:
    params: int, boolean, 
    returns: int
    locals:

class B:
  super: null
  fields:
  method run:
    params: int, int, 
    returns: int
    locals:

class Override:
  super: null
  fields:
    int x;
    int y;
    boolean f1;
    boolean f2;
  method run:
    params: int, int, 
    returns: int
    locals:
      A a;
      B b;

